### ISSUE TITLE
Bundler is scanning through nested packages incorrectly

### ISSUE THREAD

**Issue by esprehn** [facebook/react-native#17677]
### Is this a bug report?

Yes

### Have you read the [Contributing Guidelines](https://facebook.github.io/react-native/docs/contributing.html)?

Yes

### Environment

  Environment:
    OS:  macOS Sierra 10.12.6
    Node:  6.10.3
    npm:  3.10.10
    Watchman:  4.7.0

  Packages:
    react-native: 0.52.0

  Target Platform: Android

### Steps to Reproduce

We have a directory structure using lerna like:

```
packages/
  app/
    package.json
    artifacts/
      node_modules/include-common <= compiled code is here.
      app.js
    node_modules/include-common <== symlink to ../../common
    node_modules/include-webview <== symlink to ../../webview
  webview/
    node_modules/include-common => symlink to ../../common
  common/
```
  
In our project we import from include-common, and it seems the bundler is going down through the symlinks in the project and finding the typescript in common (either directly or transitively through webview) and then failing with the error:

```
While resolving module `include-common/ast/conversion`, the Haste package `include-common` was found. However the module `ast/conversion` could not be found within the package. Indeed, none of these files exist:
  * `/Checkouts/OurApp/packages/common/ast/conversion(.native||.android.js|.native.js|.js|.android.json|.native.json|.json)`
  * `/Checkouts/OurApp/packages/common/ast/conversion/index(.native||.android.js|.native.js|.js|.android.json|.native.json|.json)`
```

This is confusing since the include-common module is a sibling of app.js in the ancestor chain. If I delete BOTH symlinks inside node_modules for include-webview and include-common then things start working again.

It seems like the bundler is looking for modules depth first up through the tree instead of up the ancestor chain in node_modules directories. There's no reason that it should ever be looking inside webview here to find a nested package, but indeed it does anyway.

This worked fine in 0.51, and is broken in 0.52

### Expected Behavior

When I import from include-common it should look up all ancestors to find node_modules directories and see if there's a directory in there for include-common.

### Actual Behavior

Error from looking at directories in node_modules that are not even referenced from the code.

**Comment by esprehn** [comment]
It seems that maybe the roots computed wrong. Any symlink inside node_modules becomes a root for some reason, and then RN is searching them.

If node_modules/include-common and include-webview exist then the bundler has these roots:

```
["/Checkouts/grasshopper/packages/app",
 "/Checkouts/grasshopper/packages/common",
 "/Checkouts/grasshopper/packages/webview"]
```

but if I delete those folders then I get the roots:

```
["/Checkouts/grasshopper/packages/app"]
```

**Comment by esprehn** [comment]
Yeah so the bundler is definitely getting confused scanning symlinks in node_modules and then adding them as roots which overrides the lookup path for modules incorrectly.

This rn-cli.config.js fixes it by making the bundler stop looking through symlink roots.

```
module.exports = {
  getProjectRoots() {
    return [
      path.resolve(__dirname),
    ];
  }
};
```

**Comment by brunolemos** [comment]
Related: #17610

**Comment by t4deu** [comment]
Maybe it's related to this PR #15776

**Comment by hawkrives** [comment]
This has been resolved in one of the new metro-bundler releases, I believe.
